fs/vfs/fs_rename.c

```diff
@@ -472,6 +473,14 @@ int rename(FAR const char *oldpath, FAR const char *newpath)
       goto errout;
     }
 
+  /* Ensure that oldpath actually exists */
+
+  ret = nx_stat(oldpath, &stat_buf, 0);
```

Comment:
move the check to mountptrename, so we don't search the old patch twice

Reply:
I understand the state of problem, but not sure if this is a proper fix. I see that `mountptrename` calls `ret = oldinode->u.i_mops->stat(oldinode, newrelpath, &buf);`, so probably there is some bug in logic and that call is not hit.

Reply:
@xiaoxiang781216 You are correct about the double search.

But maybe move it after the search and before the `if (INODE_IS_MOUNTPT(oldinode) && *olddesc.relpath != '\0')` conditional? So it also protects `pseudorename()`?

Or `pseudorename()` doesn't need this check? I haven't understood this code 100% yet...

Reply:
Currently I see that fat_rename returns -2 and I'm looking for the place where file2 is deleted

Reply:
Currently I see that fat_rename returns -2 and I'm looking for the place where file2 is deleted

Reply:
Such behavior may be fat specific. Let's hold on with populating it to all fs types until we know the exact reason

Reply:
@pkarashchenko 

> I understand the state of problem, but not sure if this is a proper fix. I see that `mountptrename` calls `ret = oldinode->u.i_mops->stat(oldinode, newrelpath, &buf);`, so probably there is some bug in logic and that call is not hit.

Shouldn't a check be added like: `ret = oldinode->u.i_mops->stat(oldinode, oldrelpath, &buf);`?  
Notice the use of `oldrelpath` vs `newrelpath`.

Reply:
I was looking into this issue, but was interrupted. I'll take a look in few hours. Just want to make sure that this is a correct fix

Reply:
https://github.com/apache/incubator-nuttx/pull/7312 seems to be more localized change that performs the same. Please try it out